src/lib/array.jl

```diff
@@ -174,6 +174,7 @@ function _tryreverse(m::typeof(map), backs, Δ::Union{AbstractVector, Tuple})
 end
 _tryreverse(m, x) = x
 _tryreverse(m::typeof(map), x::Union{AbstractVector, Tuple}) = reverse(x)
+_tryreverse(m::typeof(map), backs, Δ::Nothing) = backs, fill(0,size(backs))
```

Comment:
It is also weird that we only add this for `map` not for `vmap` or `pmap`

Reply:
Also, is this meant to work when the objects being mapped over aren't subtypes of `Real`? If so, we'll need to `fill` with an element other than `0`.

Reply:
`vmap` and `pmap` can't guarantee any order of iteration, so there seems little point in reversing them.

Reply:
True, we really don't need to do any compute here technically, but it seemed like the least breaking approach here.

Agree that we can fill with the output of the forward pass type, but the thing is also that we know that they don't have gradients so maybe it just needs to return `nothing`